src/cluster_legacy.c

```diff
@@ -3515,6 +3646,39 @@ void clusterBroadcastMessage(clusterMsgSendBlock *msgblock) {
     dictReleaseIterator(di);
 }
 
+void clusterBroadcastPublishMessage(clusterMsgSendBlock *msgblock) {
+    dictIterator *di;
+    dictEntry *de;
+
+    di = dictGetSafeIterator(server.cluster->nodes);
+    while ((de = dictNext(di)) != NULL) {
+        clusterNode *node = dictGetVal(de);
+
+        if (node->flags & (CLUSTER_NODE_MYSELF | CLUSTER_NODE_HANDSHAKE)) continue;
+        if (nodeSupportsLightMsgHdr(node))
+            continue;
+        else
+            clusterSendMessage(node->link, msgblock);
+    }
+    dictReleaseIterator(di);
+}
+
+void clusterBroadcastPublishLightMessage(clusterMsgSendBlockLight *msgblock_light) {
+    dictIterator *di;
+    dictEntry *de;
+
+    di = dictGetSafeIterator(server.cluster->nodes);
+    while ((de = dictNext(di)) != NULL) {
+        clusterNode *node = dictGetVal(de);
+
+        if (node->flags & (CLUSTER_NODE_MYSELF | CLUSTER_NODE_HANDSHAKE)) continue;
+        if (nodeSupportsLightMsgHdr(node))
+            clusterSendPublishMessage(node->link, msgblock_light);
+        else
+            continue;
+    }
+    dictReleaseIterator(di);
+}
```

Comment:
I don't like this duplication of code. Like I suggested earlier can we make `clusterMsgSendBlock` opaque and typecast based on certain flag. @zuiderkwast What do you think?

Reply:
Avoid duplication, yes please. :)

Can we just use one clusterMsgSendBlock for both light and regular msg? I posted some ideas in other comments.